### PR TITLE
chore: add/remove linters for consistency

### PR DESCRIPTION
**Pull request by jkongie**
Updates the linters to be consistent with DPA standards.

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -3,46 +3,75 @@ run:
   tests: false
   modules-download-mode: readonly
 linters:
-  disable-all: true
```
Comment (jkongie) [inline .golangci.yml:6]: This was not removed in this PR because the only linter this ends up disabling is the `unused` linter.

There are a number of functions that are unused and I don't want to remove them unnecessarily.

File: .golangci.yml [hunk 2]
```diff
@@ -3,46 +3,75 @@ run:
...
-    - govet
-    - ineffassign
+    - gosimple # Default linter
+    - govet # Default linter
+    - ineffassign # Default linter
+    - intrange
+    - loggercheck
     - makezero
     - misspell
+    # - mnd
```
Comment (jkongie) [inline .golangci.yml:40]: Quite a few magic numbers exist, so this can be captured in another PR

File: .golangci.yml [hunk 3]
```diff
@@ -3,46 +3,75 @@ run:
...
+    - nlreturn
     - noctx
     - nolintlint
+    - paralleltest
+    - prealloc
     - predeclared
     - promlinter
-    - staticcheck
+    - reassign
+    # - revive
```
Comment (jkongie) [inline .golangci.yml:52]: This is a large lint, so better done in a separate PR

File: cmd/root.go [hunk 4]
```diff
@@ -4,9 +4,10 @@ import (
 	"os"
 
 	"github.com/rs/zerolog"
-	"github.com/smartcontractkit/timelock-worker/pkg/logger"
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
+
+	"github.com/smartcontractkit/timelock-worker/pkg/logger"
```
Comment (jkongie) [inline cmd/root.go:10]: Fixes import ordering

File: tests/integration/suite.go [hunk 5]
```diff
@@ -42,10 +42,6 @@ func (s *integrationTestSuite) Logf(format string, args ...any) {
 	s.T().Logf(format, args...)
 }
 
-func (s *integrationTestSuite) Run(name string, subtestFunc func(t *testing.T)) bool {
-	return s.T().Run(name, subtestFunc)
-}
-
```
Comment (jkongie) [inline tests/integration/suite.go:48]: This didn't seem to be needed but please correct me if there was a reason to wrap this

File: tests/integration/suite.go [hunk 6]
```diff
@@ -80,9 +76,10 @@ func (s *integrationTestSuite) DeployTimelock(
 
 	receipt, err := bind.WaitMined(ctx, client, transaction)
 	s.Require().NoError(err)
-	s.Require().Equal(receipt.Status, types.ReceiptStatusSuccessful)
```
Comment (jkongie) [inline tests/integration/suite.go:83]: Incorrect ordering

**Review by gustavogama-cll — APPROVED** [review]
